src/pages/Market/components/AllItem.tsx

```diff
@@ -0,0 +1,114 @@
+    } else if (width < 1280) {
+        return 6;
+    } else {
+        return 10;
+    }
+};
+
+interface Item {
+    id: number;
+    name: string;
+    description: string;
+    price: number;
+    tags: string[];
+    images: string[];
+    ownerId: number;
+    ownerNickname: string;
+    favoriteCount: number;
+    createdAt: string;
+}
+
+interface ItemList {
+    totalCount: number;
+    list: Item[];
+}
+
+function AllItem() {
+    const [itemList, setItemList] = useState<ItemList>({ totalCount: 0, list: [] });
+    const [pageSize, setPageSize] = useState(getPageSize());
+    const [order, setOrder] = useState('recent');
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+    const handleNewestClick = () => {
+        setOrder('recent');
+        setIsDropdownOpen(false);
+    };
+
+    const handleFavoriteClick = () => {
+        setOrder('favorite');
+        setIsDropdownOpen(false);
+    };
+
+    const fetchSortedData = useCallback(async () => {
+        try {
+            const products = await getProducts({ orderBy: order, pageSize });
+            setItemList(products); 
+        } catch (error) {
+            console.error('Error fetching products:', error);
+        }
+    }, [order, pageSize]);
+
+    useEffect(() => {
+        fetchSortedData(); 
+    }, [fetchSortedData]);
+
+    useEffect(() => {
+        const handleResize = () => {
+            setPageSize(getPageSize());
+            fetchSortedData(); 
+        };
+
+        window.addEventListener('resize', handleResize);
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
+    }, [fetchSortedData]);
+
+    const toggleDropdown = () => {
+        setIsDropdownOpen(!isDropdownOpen);
+    };
+
+    return (
+        <div className={styles.allItem}>
+            <div className={styles.allItemContainer}>
+                <div className={styles.allItemHeader}>
+                    <h3 className={styles.listTitle}>전체 상품</h3>
+                    <div className={styles.allItemHeaderRight}>
+                        <div className={styles.searchContainer}>
+                            <img className={styles.searchIcon} src={SearchIcon} alt="검색" />
+                            <input className={styles.searchInput} placeholder="검색할 상품을 입력해주세요" />
+                        </div>
+                        <Link className={styles.addItemButton} to="/addItem">상품 등록하기</Link>
+                        <div className={styles.dropdownContainer}>
+                            <button className={styles.dropdownButton} onClick={toggleDropdown}>
+                                {order === 'recent' ? '최신순' : '인기순'} <span className={styles.dropdownArrow}>▼</span>
+                            </button>
+                            <ul className={`${styles.dropdownMenu} ${isDropdownOpen ? styles.show : ''}`}>
+                                <li className={styles.dropdownOption} onClick={handleNewestClick}>최신순</li>
```

Comment:
P1:
li 태그보다 button 같은 것을 사용해주세요~
이렇게 해주시면 handleNewestClick, handleFavoriteClick 가 아니라 하나의 함수로 사용할 수 있습니다.

```suggestion
                                <li className={styles.dropdownOption}>
                                  <button type='button' value='recent' onClick={handleNewestClick}>최신순</button>
                                </li>
```